### ISSUE TITLE
Outdated stack usage string

### ISSUE THREAD

**Issue by rimbaudcode** [commercialhaskell/stack#3687]
When I query `stack` on how to start a new project with a specific resolver, I get:

```
$ stack new

Missing: PACKAGE_NAME

Usage: stack new PACKAGE_NAME [--bare] [TEMPLATE_NAME] [-p|--param KEY:VALUE]
                 [DIRS] [--solver] [--omit-packages] [--force]
                 [--ignore-subdirs] [--help]
  Create a new project from a template. Run `stack templates' to see available
  templates.
```

But, when I want to actually start a project, say with the lts-9.6 resolver, I follow the usage suggested by `stack` and hit an error:

```
$ cd sandbox
$ stack new hello simple --solver lts-9.6
Downloading template "simple" to create project "hello" in hello/ ...
~/sandbox/lts-9.6: canonicalizePath: does not exist (No such file or directory)

```

However, if I use `--resolver lts-9.6` instead of `--solver lts-9.6`, everything runs smoothly:

```
$ stack new hello simple --resolver lts-9.6
Downloading template "simple" to create project "hello" in hello/ ...
Looking for .cabal or package.yaml files to use to init the project.
Using cabal packages:
- hello/hello.cabal

Selected resolver: lts-9.6
Initialising configuration using resolver: lts-9.6
Total number of user packages considered: 1
Writing configuration to file: hello/stack.yaml
All done.
```

This means, I guess, that stack's `usage` string is outdated (?)

I'm using:
* stack Version 1.2.0 x86_64 hpack-0.14.0
* stack installed with Mac's homebrew
* ghc version 8.2.2
* macOS High Sierra 10.13.2 (17C88)

Cheers!

**Comment by kadoban** [comment]
That's a pretty old version of stack. You're likely to run into some trouble pretty quickly, I'd recommend you upgrade. The easy way is usually via running `stack upgrade`.

This particular part hasn't changed though.

While the help text is a bit weird in a couple of ways, the part you're talking about I think comes from a misunderstanding. The `--resolver` flag, which you use to specify something like lts-9.6 doesn't appear in the help text. This is an unfortunate consequence of `--resolver` being an acceptable option to a whole bunch of different commands, so it's handled a bit differently than others. If you do just `stack --help` you should see it near the top.

It really should be there in commands like this too, ideally, but it's not.

But `--solver` is separate, and doesn't take an argument like that. So the lts-9.6 part ends up having nothing to do with the `--solver` argument. I'm not sure actually what exactly it's doing there, I think that ends up being the `[DIR]` from the help text, which I'm not quite clear what that does. I think that might be an option from the related command `stack init` that kind of bleeds over to this one. I'm unsure if it has a real use-case when specified for `stack new`

**Comment by rimbaudcode** [comment]
Thanks for the advice on upgrading `stack` and the reminder on `stack`'s distinction between `--solver` and `--resolver`. It's been a while since I used `stack` and I forgot that subtlety.

However, if starting a new project with a specific resolver is required, then `stack new` does not display this useful option. I could not even find it in the official docs or some tutorials I checked. I just found it by trial and error out of a need to get around an issue.
